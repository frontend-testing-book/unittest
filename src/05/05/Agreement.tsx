import { Checkbox } from "./Checkbox";

type Props = {
  onChange?: (checked: boolean) => void;
};

export const Agreement = ({ onChange }: Props) => {
  return (
    <fieldset>
      <legend>利用規約の同意</legend>
      <label>
        <Checkbox name="agree" onChange={onChange} />
        当サービスの<a href="/terms">利用規約</a>を確認し、これに同意します
      </label>
    </fieldset>
  );
};
