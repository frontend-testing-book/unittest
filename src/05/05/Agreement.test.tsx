import { render, screen } from '@testing-library/react'

import { Agreement } from './Agreement'

test('fieldset のアクセシブルネームは、legend を引用している', () => {
  render(<Agreement />)
  expect(
    screen.getByRole('group', { name: '利用規約の同意' }),
  ).toBeInTheDocument()
})

test('チェックボックスはチェックが入っていない', () => {
  render(<Agreement />)
  expect(screen.getByRole('checkbox')).not.toBeChecked()
})

test('利用規約へのリンクがある', () => {
  render(<Agreement />)
  expect(screen.getByRole('link')).toBeInTheDocument()
  expect(screen.getByRole('link')).toHaveTextContent('利用規約')
  expect(screen.getByRole('link')).toHaveAttribute('href', '/terms')
  expect(screen.getByRole('link', { name: '利用規約' })).toHaveAttribute(
    'href',
    '/terms',
  )
})

test('Snapshot: 利用規約の同意が表示される', () => {
  const { container } = render(<Agreement />)
  expect(container).toMatchSnapshot()
})
