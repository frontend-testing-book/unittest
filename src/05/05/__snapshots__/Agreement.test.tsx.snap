// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: 利用規約の同意が表示される 1`] = `
<div>
  <fieldset>
    <legend>
      利用規約の同意
    </legend>
    <label>
      <span>
        <input
          name="agree"
          style="display: none;"
          type="checkbox"
        />
        <span
          style="display: inline-block; text-align: center; width: 24px; height: 24px; border-radius: 4px; cursor: pointer; color: rgb(255, 255, 255); background-color: rgb(204, 204, 204);"
        >
          ✔︎
        </span>
      </span>
      当サービスの
      <a
        href="/terms"
      >
        利用規約
      </a>
      を確認し、これに同意します
    </label>
  </fieldset>
</div>
`;
